website/integrations: add osticket (#16572)
goauthentik/authentik: website/integrations/infrastructure/osticket/index.md

---
title: Integrate with osTicket
sidebar_label: osTicket
support_level: community
---

## What is osTicket

> osTicket is a web-based, open source user support/ticketing solution.

> > Download and install it on your own servers. For free.

Reviewer: ```suggestion
```

> > Download and install it on your own servers. For free.

Reviewer 2: Let's remove this line please

>
> -- https://osticket.com

## Preparation

The following placeholders are used in this guide:

- `osticket.company` is the FQDN of the osTicket installation.
- `authentik.company` is the FQDN of the authentik installation.

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

## authentik configuration

To support the integration of osTicket with authentik, you need to create an application/provider pair in authentik.

### Create an application and provider in authentik

> 1. Log in to authentik as an administrator and open the authentik Admin interface.

Reviewer: ```suggestion
1. Log in to authentik as an administrator, and open the authentik Admin interface.
```

    - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Take note of the **slug** as it will be required later.
    - **Choose a Provider type**: select **OAuth2/OpenID Connect** as the provider type.
    - **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
        - Note the **Client ID** and **slug** values because they will be required later.
        - Set a `Strict` redirect URI to `https://osticket.company/osticket/api/auth/oauth2`.
        - Select any available signing key.
        - Under **Advanced Protocol Settings**:
            - **Subject Mode**: `Based on the User's Email`
    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

## osTicket configuration

> To configure the osTicket OAuth2 first download the plugin from [the osTicket website](https://osticket.com/download) and place it into the `include/plugins` folder. Then go to the **Admin Panel** (link top right) and select **Manage** > **Plugins**. Next, click Add New Plugins and follow the onscreen instructions to install Oauth2 client.

Reviewer 2: ```suggestion
1. To configure osTicket with authentik, you must download the Oauth2 plugin from the [osTicket website](https://osticket.com/download) and place it in the `include/plugins` folder of your installation.
```

> 

Reviewer 2: Let's remove this new line

> Once installed click on the new Oauth2 client plug in, set Status to Active and click on Save Changes.

Reviewer 2: ```suggestion
2. Then, go to the osTicket administration interface and navigate to **Manage** > **Plugins**. Follow the on-screen instructions to install the plugin.
3. Once the install process is complete, click on the newly added plugin, set the **Status** to `Active` and click **Save Changes**.
```

> 

Reviewer 2: Let's remove this new line please

> Then go to "Instances", click on "Add New Instance" and select "OAuth2 - Other"

Reviewer 2: ```suggestion
4. Navigate to **Instances**, click on **Add New Instance** and select **OAuth2 - Other**.
```

> 

Reviewer 2: Let's remove this new line please

> - On the **Instance** tab fill in the Name and set Status to Enabled

Reviewer 2: ```suggestion
5. Set the following required configurations:
	- In the **Instance** tab:
		- **Name**: `authentik`
		- **Status**: `Enabled`
```

> - On the **Config** tab change the following fields

Reviewer 2: ```suggestion
  - In the **Config** tab:
```

>     - **Name**: `Authentik`

Reviewer 2: ```suggestion 
    - **Name**: `authentik`
```

    - **Authentication Target**: `Agents Only` or `End Users Only` or `Agents and End Users`

>     - **Authentication Label**: `Authentik`

Reviewer 2: ```suggestion
    - **Authentication Label**: `authentik`
```

    - **Redirect URI**: `https://osticket.company/osticket/api/auth/oauth2`

>     - **Client Id**: `<Client ID>`

Reviewer 2: ```suggestion
    - **Client Id**: Set the Client ID from authentik
```

>     - **Client Secret**: `<Client Secret>`

Reviewer 2: ```suggestion
    - **Client Secret**: Set the Client Secret from authentik
```

    - **Authorization Endpoint**: `https://authentik.company/application/o/authorize/`
    - **Token Endpoint**: `https://authentik.company/application/o/token/`

>     - **Resource Details Endpoint**: `https://authentik.company`/application/o/userinfo/

Reviewer 2: ```suggestion
    - **Resource Details Endpoint**: `https://authentik.company/application/o/userinfo/`
```

    - **Scopes**: `email openid profile`

> - click on Save Changes

Reviewer 2: ```suggestion
6. Click **Save Changes**.
```

> To configure the osTicket OAuth2 first download the plugin from [the osTicket website](https://osticket.com/download) and place it into the `include/plugins` folder. Then go to the **Admin Panel** (link top right) and select **Manage** > **Plugins**. Next, click Add New Plugins and follow the onscreen instructions to install Oauth2 client.
> 
> Once installed click on the new Oauth2 client plug in, set Status to Active and click on Save Changes.
> 
> Then go to "Instances", click on "Add New Instance" and select "OAuth2 - Other"
> 
> - On the **Instance** tab fill in the Name and set Status to Enabled
> - On the **Config** tab change the following fields
>     - **Name**: `Authentik`
>     - **Authentication Target**: `Agents Only` or `End Users Only` or `Agents and End Users`
>     - **Authentication Label**: `Authentik`
>     - **Redirect URI**: `https://osticket.company/osticket/api/auth/oauth2`
>     - **Client Id**: `<Client ID>`
>     - **Client Secret**: `<Client Secret>`
>     - **Authorization Endpoint**: `https://authentik.company/application/o/authorize/`
>     - **Token Endpoint**: `https://authentik.company/application/o/token/`
>     - **Resource Details Endpoint**: `https://authentik.company`/application/o/userinfo/
>     - **Scopes**: `email openid profile`
> 
> - click on Save Changes

Reviewer: ```suggestion
1. Download the OAuth2 plugin from [the osTicket website](https://osticket.com/download) and place it into the `include/plugins` folder of your osTicket installation. Then 
2. Login to osTicket as an administrator and navigate to the **Admin Panel** > **Manage** > **Plugins**. 
3. Click **Add New Plugins** and follow the onscreen instructions to install the OAuth2 plugin.
4. Once installed click on the newly installed OAuth2 plugin, set **Status** to **Active ** and then click **Save Changes**.
5. Navigate to **Instances**, click on **Add New Instance**, select **OAuth2 - Other** and set the following fields:
    - On the **Instance** tab:
        - Define the **Name**.
        - Set **Status** to `Enabled`.
    - On the **Config** tab:
        - **Name**: `authentik`
        - **Authentication Target**: `Agents Only` or `End Users Only` or `Agents and End Users`
        - **Authentication Label**: `authentik`
        - **Redirect URI**: `https://osticket.company/osticket/api/auth/oauth2`
        - **Client Id**: <client_id>
        - **Client Secret**: <client_secret>
        - **Authorization Endpoint**: `https://authentik.company/application/o/authorize/`
        - **Token Endpoint**: `https://authentik.company/application/o/token/`
        - **Resource Details Endpoint**: `https://authentik.company/application/o/userinfo/`
        - **Scopes**: `email openid profile`

6. Click on **Save Changes**.
```

> ## Additional Resources

Reviewer 2: Should be called just Resources and formatted according to the template: https://github.com/goauthentik/authentik/blob/main/website/integrations/template/service.md#references (ignore the bad header for now, it needs to be changed)

> ## Additional Resources

Reviewer: ```suggestion
## Resources
```

> - https://docs.osticket.com/en/latest/Guides/OAuth2%20Guide.html

Reviewer: ```suggestion
- [osTicket Docs - OAuth2 Guide](https://docs.osticket.com/en/latest/Guides/OAuth2%20Guide.html)
```

    - **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
        - Note the **Client ID** and **slug** values because they will be required later.
        - Set a `Strict` redirect URI to `https://osticket.company/osticket/api/auth/oauth2`.
        - Select any available signing key.
        - Under **Advanced Protocol Settings**:
            - **Subject Mode**: `Based on the User's Email`
    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

## osTicket configuration

1. Download the OAuth2 plugin from [the osTicket website](https://osticket.com/download) and place it into the `include/plugins` folder of your osTicket installation.
2. Log in to osTicket as an administrator and navigate to the **Admin Panel**
3. On the **Admin Panel**, go to **Manage** > **Plugins**.

> 4. Click **Add New Plugins** and follow the onscreen instructions to install the OAuth2 plugin.

Reviewer 2: ```suggestion
4. Click **Add New Plugins** and follow the on-screen instructions to install the OAuth2 plugin.
```

Apparently both are fine, but this seems to be the preferred way of writing it

> 5. Once installed click on the newly installed OAuth2 plugin, set **Status** to **Active ** and then click **Save Changes**.

Reviewer 2: ```suggestion
5. Once installed, click on the newly installed OAuth2 plugin, set **Status** to `Active`, then click **Save Changes**.
```

> 6. Navigate to **Instances**, click on **Add New Instance**, select **OAuth2 - Other** and set the following fields:

Reviewer 2: ```suggestion
6. Navigate to **Instances**, click on **Add New Instance**, select **OAuth2 - Other**, and set the following fields:
```

    - On the **Instance** tab:

>         - Define the **Name**.

Reviewer 2: ```suggestion
        - Set a **Name**.
```

---
title: Integrate with osTicket
sidebar_label: osTicket
support_level: community
---

## What is osTicket

> osTicket is a web-based, open source user support/ticketing solution.
>
> -- https://osticket.com

## Preparation

The following placeholders are used in this guide:

- `osticket.company` is the FQDN of the osTicket installation.
- `authentik.company` is the FQDN of the authentik installation.

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

> ## Authentik configuration

Reviewer 2: ```suggestion
## authentik configuration
```

Author: wait it should be lowercase?

Author: guess i was too trigger happy xd

---
title: Integrate with osTicket
sidebar_label: osTicket
support_level: community
---

## What is osTicket

> osTicket is a web-based, open source user support/ticketing solution.
>
> -- https://osticket.com

## Preparation

The following placeholders are used in this guide:

- `osticket.company` is the FQDN of the osTicket installation.
- `authentik.company` is the FQDN of the authentik installation.

:::note
This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
:::

## authentik configuration

To support the integration of osTicket with authentik, you need to create an application/provider pair in authentik.

> ### Create an application and provider in authentik

Reviewer 3: also thanks for this contribution. ;-)

> ### Create an application and provider in authentik

Reviewer 2: Wait. Don't we always have this header

> ### Create an application and provider in authentik

Author: also caught the all lower case one too haha
